editions/tw5.com/tiddlers/filters/syntax/Filter Expression.tid

```diff
@@ -1,73 +1,24 @@
-** See [[Map Filter Run Prefix]]. <<.from-version "5.2.0">>
-
-
-<<.tip "Compare named filter run prefix `:filter` with [[filter Operator]] which applies a subfilter to every input title, removing the titles that return an empty result from the subfilter">>
-
-<<.tip "Compare named filter run prefix `:reduce` with [[reduce Operator]] which is used to used to flatten a list of items down to a single item by repeatedly applying a subfilter.">> 
-
-<<.tip """Within the filter runs prefixed with `:reduce`, `:sort`, `:map` and `:filter`, the "currentTiddler" variable is set to the title of the tiddler being processed. The value of currentTiddler outside the subfilter is available in the variable "..currentTiddler".<<.from-version "5.2.0">>""" >>
+Parents: <$macrocall $name=".recursiveParentTag" current=<<currentTiddler>> stop="Filters"/>
 
-In technical / logical terms:
+A <<.def "filter expression">> is the outermost level of the [[filter syntax|Filter Syntax]]. It consists of optional filter run [[prefix|Filter Run Prefix]] and a filter [[run|Filter Run]]. Multiple filter runs are separated by [[whitespace|Filter Whitespace]].
 
-|!Run |!Equivalent named prefix |!Interpretation |!Output |
-|`run` |`:or[run]` |de-duplicated union of sets |... OR run |
-|`=run` |`:all[run]` |union of sets without de-duplication |... OR run |
-|`+run` |`:and[run]` |accumulation of filter steps |... AND run |
-|`-run` |`:except[run]` |difference of sets |... AND NOT run |
-|`~run` |`:else[run]` |else |... ELSE run |
-||`:intersection`|intersection of sets||
+The first run usually has //no// filter run prefix
 
-For the difference between `+` and `:intersection`, see [[Filter Run Prefix (Examples)]].
-
-The input of a run is normally a list of all the non-[[shadow|ShadowTiddlers]] tiddler titles in the wiki (in no particular order). But the `+` prefix can change this:
-
-|Prefix|Input|h
-|`-`, `~`, `=`, `:intersection` or none| <$link to="all Operator">`[all[]]`</$link> tiddler titles, unless otherwise determined by the first [[filter operator|Filter Operators]]|
-|`+`, `:filter`, `:reduce`,`:sort`|the filter output of all previous runs so far|
-
-Precisely because of varying inputs, be aware that both prefixes `-` and `+` do not behave inverse to one another!
-
-For example, in both of the following, `$:/baz` will only be removed if it actually exists:
-
-* <$link to="is Operator"> `foo bar $:/baz -[is[system]]`</$link>
-* <$link to="prefix Operator">`foo bar $:/baz -[prefix[$:/]]`</$link>
-
-To understand why, consider the input for both final runs with their `-` prefix.
-
-In order to remove `$:/baz` in any case, existing or not, simply use the `+` prefix with [[negated filter operators|Filter Operators]]:
+<$railroad text="""
+\start none
+{  (
+     - |
+     : [[<"prefix">|"Filter Run Prefix"]]
+   )
+   [[run|"Filter Run"]]  
+   +  [[whitespace|"Filter Whitespace"]]
+}
+"""/>
```

Comment:
The start is open, which indicates that users should look up one level. Ideally the plugin would allow a "label" with a link up. .. But since we do not have that yet I used it that way. .. after the [run] box it's always terminated.

But I did see some other diagrams where I did miss the "none" pragma.

Reply:
> after the [run] box it's always terminated

@pmario: Yes, but the termination happens in "Filter Syntax", not here.
I believe we need a general concensus on how the railroad diagrams should work. Should they only depict the contents of the current tiddler, so they can replace the corresponding items in a higher level, or should they show more context? Maybe @Jermolene can offer some insights on his thinking when making the other ones?